### PR TITLE
add can_execute_individually to ExternalAssetCheck

### PR DESCRIPTION
**Pull request by johannkm**
Currently we let you try to execute a check individually on a non subsettable assets def, then return a invalid subset error. This diff will let us preemptively disable the execute button with a tooltip. It will also help with the (not yet landed) launchpad toggle for including or not including checks. We can specify that the toggle will exclude xyz checks, but leave in zxy other checks.

### CODE REVIEW

**Comment by johannkm** [comment]
Current dependencies on/for this PR:
* master
  * **PR #16711** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/16711" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/dagster-io/dagster/16711?utm_source=stack-comment).

File: python_modules/dagster/dagster/_core/host_representation/external_data.py [hunk 1]
```diff
@@ -1091,6 +1091,12 @@ def __new__(
         )
 
 
+@whitelist_for_serdes
+class AssetCheckCanExecuteIndividually(enum.Enum):
+    TRUE = "TRUE"
+    CANNOT_SUBSET = "CANNOT_SUBSET"
```
Comment (johannkm) [inline python_modules/dagster/dagster/_core/host_representation/external_data.py:1097]: might there be other future reasons? maybe can cut this down to a bool

**Comment by sryza** [comment]
I noticed that we don't have a parameter with the same name on `ExternalAssetNode` - how do we handle it in that case? However we do it there, would it make the sense to do the same here?

Maybe via the `atomic_execution_unit_id`?

**Comment by johannkm** [comment]
Yeah for assets we use atomic_execution_unit_id for the asset daemon, and let users hit the subset error for the materialize button. We could port over `atomic_execution_unit_id`- I think the implication is that we'd need to construct an asset graph for the fetch_asset_check method https://github.com/dagster-io/dagster/blob/rl/allow-subsetted-asset-checks/python_modules/dagster-graphql/dagster_graphql/implementation/fetch_asset_checks.py#L28. Not sure how slow that is, probably fine once we do batching

**Comment by sryza** [comment]
> We could port over atomic_execution_unit_id

I think we're likely going to need the `atomic_execution_unit_id` for backfills and auto-materialize, so worth adding.

> I think the implication is that we'd need to construct an asset graph for the fetch_asset_check method

If we're concerned about performance, could we do the analogous thing to "let users hit the subset error for the materialize button" and let them hit the subset error when they try to execute the check?

**Comment by johannkm** [comment]
> could we do the analogous thing to "let users hit the subset error for the materialize button" and let them hit the subset error when they try to execute the check?

yeah we already do this, was just hoping to improve that experience, plus help the 'include checks' toggle. I'll switch to atomic_execution_unit_id, and if perf is an issue we can always consider doing something like this too

**Comment by johannkm** [comment]
Tested ` ExternalAssetGraph.from_workspace(` perf, it's < 0.1 sec on our toys workspace so should be fine

**Pull request closed without merging**